hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -146,33 +146,26 @@ checkAbort ctx@ScriptContext{scriptContextTxInfo = txInfo} headCurrencySymbol pa
 --
 --   * All participants have committed (even empty commits)
 --
---   * All commits are properly collected and locked into the contract as a hash
+--   * All commits are properly collected and locked into η as a hash
 --     of serialized tx outputs in the same sequence as commit inputs!
 --
 --   * The transaction is performed (i.e. signed) by one of the head participants
 --
 --   * State token (ST) is present in the output
---
--- It must also initialize the on-chain state η* with a snapshot number and a
--- hash of committed outputs.
---
--- (*) In principle, η contains not a hash but a full UTXO set as well as a set
--- of dangling transactions. However, in the coordinated version of the
--- protocol, there can't be any dangling transactions and thus, it is no longer
--- required to check applicability of those transactions to the UTXO set. It
--- suffices to store a hash of the resulting outputs of that UTXO instead.
 checkCollectCom ::
   -- | Script execution context
   ScriptContext ->
   (ContestationPeriod, [Party], CurrencySymbol) ->
   Bool
 checkCollectCom ctx@ScriptContext{scriptContextTxInfo = txInfo} (contestationPeriod, parties, headId) =
-  mustNotMintOrBurn txInfo
-    && mustCollectUtxoHash
+  mustCollectUtxoHash
     && mustNotChangeParameters
+    && mustCollectAllValue
+    -- XXX: Is this really needed? If yes, why not check on the output?
```

Comment:
Decision: Checking for the ST should be part of the `mustCollectAllValue` rule (to match the spec) and we hence should check that there is an ST in the output.

Conretely, this means that we need to put a `hasST` check to `mustCollectAllValue` because the current clause (the inverse of the thing in the spec) does not cover it yet.